File: HSEvents.Web/client/models/address.models.ts
export class Address {
    id: number;
    caption: string;
    shortName: string;
    house: string;
    streetId: number;
}

export class Street {
    id: number;
    name: string;
    cityId: number;
    areaName: string;
}

export class City {
    id: number;
    name: string;
    cityType: CityType;
    regionId: number;
    areaName: string;
}

export class CityType {
    id: number;
    name: string;
    shortName: string;
}

export class Region {
    id: number;
    name: string;
    country: Country;
}

export class Country {
    id: number;
    name: string;
}

File: Infrastructure/Repositories/Dto/AddressModels.cs

using Domain;

namespace Infrastructure.Repositories.Dto
{
    public class AddressDto
    {
        public long Id { get; set; }
        public string Caption { get; set; }
        public string ShortName { get; set; }
        public string House { get; set; }
        public long StreetId { get; set; }
    }

    public class StreetDto
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public City City { get; set; }
        public string AreaName { get; set; }
    }

    public class CityDto
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public CityType CityType { get; set; }
        public Region Region { get; set; }
        public string AreaName { get; set; }
    }
}